### PR TITLE
5666 refactor domains

### PR DESCRIPTION
**Pull request by codyseibert**
- If the terraform state bucket changes (i.e ZONE_NAME !== previous EFCMS_DOMAIN)
  - create an s3 bucket called "${ZONE_NAME}.terraform.deploys"
  - copy the documents-$ENV.tfstate, permissions-$ENV.tfstate, and ui-$ENV.tfstate files from the previous deploys bucket into new bucket
- Setup circle ci environment variables
  - EFCMS_DOMAIN_DEV = dev.ef-cms.ustaxcourt.gov
  - EFCMS_DOMAIN_STG = stg.ef-cms.ustaxcourt.gov
  - EFCMS_DOMAIN_TEST = test.ef-cms.ustaxcourt.gov
  - EFCMS_DOMAIN_IRS = irs.ef-cms.ustaxcourt.gov
  - EFCMS_DOMAIN_MIG = mig.ef-cms.ustaxcourt.gov
  - EFCMS_DOMAIN_PROD = dawson.ustaxcourt.gov
  - ZONE_NAME_DEV = ef-cms.ustaxcourt.gov
  - ZONE_NAME_STG = ef-cms.ustaxcourt.gov
  - ZONE_NAME_TEST = ef-cms.ustaxcourt.gov
  - ZONE_NAME_IRS = ef-cms.ustaxcourt.gov
  - ZONE_NAME_MIG = ef-cms.ustaxcourt.gov
  - ZONE_NAME_PROD = dawson.ustaxcourt.gov
- Create the following buckets
  - $ENV.ustc-case-mgmt.flexion.us-documents-$ENV-us-east-1
  - $ENV.ustc-case-mgmt.flexion.us-documents-$ENV-us-west-1
  - $ENV.ustc-case-mgmt.flexion.us-temp-documents-$ENV-us-east-1
  - $ENV.ustc-case-mgmt.flexion.us-temp-documents-$ENV-us-west-1
- Copy over the files from old buckets to new buckets
  - aws s3 sync s3://ustc-case-mgmt.flexion.us-documents-$ENV-us-east-1 s3://$ENV.ustc-case-mgmt.flexion.us-temp-documents-$ENV-us-east-1
  - aws s3 sync s3://ustc-case-mgmt.flexion.us-documents-$ENV-us-west-1 s3://$ENV.ustc-case-mgmt.flexion.us-temp-documents-$ENV-us-west-1
  - aws s3 sync s3://ustc-case-mgmt.flexion.us-temp-documents-$ENV-us-east-1 s3://$ENV.ustc-case-mgmt.flexion.us-temp-documents-$ENV-us-east-1
  - aws s3 sync s3://ustc-case-mgmt.flexion.us-temp-documents-$ENV-us-west-1 s3://$ENV.ustc-case-mgmt.flexion.us-temp-documents-$ENV-us-west-1
- Empty the following buckets, where $ENV is the environment you are updating (dev, stg, exp1, irs...)
  - ustc-case-mgmt.flexion.us.efcms.$ENV.us-east-1.deploys
  - ustc-case-mgmt.flexion.us.efcms.$ENV.us-west-1.deploys
  - failover-ui-$ENV.ustc-case-mgmt.flexion.us
  - failover-ui-public-$ENV.ustc-case-mgmt.flexion.us
  - ui-public-$ENV.ustc-case-mgmt.flexion.us
  - ui-$ENV.ustc-case-mgmt.flexion.us
  - ustc-case-mgmt.flexion.us-documents-$ENV-us-east-1
  - ustc-case-mgmt.flexion.us-temp-documents-$ENV-us-east-1
- Delete the following cloudfront distributions manually
  - ui-public-$ENV.ustc-case-mgmt.flexion.us
  - ui-$ENV.ustc-case-mgmt.flexion.us
- Go into the gateway_api_$ENV and delete the /clamav resource

### CODE REVIEW

File: get-keys.sh [hunk 1]
```diff
@@ -29,6 +29,8 @@ elif [[ $BRANCH == 'test' ]] ; then
   echo "${DYNAMSOFT_PRODUCT_KEYS_TEST}"
 elif [[ $BRANCH == 'migration' ]] ; then
   echo "${DYNAMSOFT_PRODUCT_KEYS_MIG}"
+elif [[ $BRANCH == 'dawson' ]] ; then
+  echo "${DYNAMSOFT_PRODUCT_KEYS_EXP}"
```
Comment (rachaelparris) [inline get-keys.sh]: Should this be `DYNAMSOFT_PRODUCT_KEYS_DAWSON`?

File: web-api/src/app.js [hunk 2]
```diff
@@ -657,6 +657,10 @@ app.post('/cases', lambdaWrapper(createCaseLambda));
  * documents
  */
 app.post('/documents/:documentId/validate', lambdaWrapper(validatePdfLambda));
+app.post(
+  '/clamav/documents/:documentId/virus-scan',
+  lambdaWrapper(virusScanPdfLambda),
+);
```
Comment (rachaelparris) [inline web-api/src/app.js]: It looks like this is a duplicate?

File: get-efcms-domain.sh [hunk 3]
```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+# Returns the environment associated with each branch
+
+# Usage
+#   ./get-env.sh develop
```
Comment (rachaelparris) [inline get-efcms-domain.sh]: Documentation needs to be updated

File: get-zone-name.sh [hunk 4]
```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+# Returns the environment associated with each branch
+
+# Usage
+#   ./get-env.sh develop
```
Comment (rachaelparris) [inline get-zone-name.sh]: This needs a documentation update too.

File: web-api/terraform/template/cognito.tf [hunk 5]
```diff
@@ -20,9 +20,9 @@ resource "aws_cognito_user_pool" "pool" {
   admin_create_user_config {
     allow_admin_create_user_only = false
     invite_message_template {
-      sms_message = "Your username is {username} and temporary password is {####}."
+      sms_message   = "Your username is {username} and temporary password is {####}."
       email_subject = "Update Your Email with the U.S. Tax Court"
-      email_message = "Welcome to the U.S. Tax Court case management system. You are now able to log in to view and manage your cases.<br /><br /><b>Your username: </b>{username}<br /></br /><b>Temporary password: </b>{####}<br /><br /><br />For added security, please log in to the <a href='https://ui-${var.environment}.${var.dns_domain}/'>U.S. Tax Court site</a> to change your password."
+      email_message = "Welcome to the U.S. Tax Court case management system. You are now able to log in to view and manage your cases.<br /><br /><b>Your username: </b>{username}<br /></br /><b>Temporary password: </b>{####}<br /><br /><br />For added security, please log in to the <a href='https://${var.dns_domain}/'>U.S. Tax Court site</a> to change your password."
```
Comment (rachaelparris) [inline web-api/terraform/template/cognito.tf]: Should this be `app.${var.dns_domain}`?

File: web-client/pa11y-ci-smoketests.config.js [hunk 6]
```diff
@@ -1,10 +1,10 @@
-const { EFCMS_DOMAIN, ENV, PETITIONS_CLERK_TOKEN } = process.env;
+const { EFCMS_DOMAIN, PETITIONS_CLERK_TOKEN } = process.env;
 
 const urls = [
   {
     actions: ['wait for .big-blue-header to be visible'],
     notes: 'verifies that the user can login and the UI is loaded',
-    url: `http://ui-${ENV}.${EFCMS_DOMAIN}/log-in?token=${PETITIONS_CLERK_TOKEN}&path=/`,
+    url: `http://${EFCMS_DOMAIN}/log-in?token=${PETITIONS_CLERK_TOKEN}&path=/`,
```
Comment (rachaelparris) [inline web-client/pa11y-ci-smoketests.config.js]: Should this be `app.${EFCMS_DOMAIN}`? I think I'm not following why sometimes `app` is included and sometimes it's not. Is this an oversight, or on purpose?

File: web-api/terraform/template/cognito.tf (hunk 5 above)
Comment (codyseibert) [reply web-api/terraform/template/cognito.tf]: app is the private application, so yup you are right

Comment (rachaelparris) [reply web-api/terraform/template/cognito.tf]: So is just `{domain}` the public site, and `app.{domain}` is the application that we log in to?

**Review by rachaelparris — APPROVED** [review]

File: web-api/terraform/template/cognito.tf (hunk 5 above)
Comment (codyseibert) [reply web-api/terraform/template/cognito.tf]: correct

**Pull request merged**